### PR TITLE
feature: fix take detection ui

### PR DESCRIPTION
**Pull request by MutazAshhab**
Depends on https://github.com/chloebrett/mlvet/pull/265

Updates the UI to match the figma mock ups

What it looks like:

https://user-images.githubusercontent.com/86769131/190136301-8d6c84ed-7431-4dda-99b7-47ca43c706b6.mp4

------
### OS

- [X] Linux
- [ ] MacOS
- [ ] Windows

### CODE REVIEW

**Review by emmapainter — APPROVED** [review]
UI looking good! This knocks off 6 tickets which is awesome. I'm only approving the functionality/UI as a PM - wait for an actual code review before merging :)

**Comment by ghost** [comment]
For future reference, I believe the GIT practice is: 
When a PR is dependent on another open PR, rather than merge into develop, merge into that branch.
That way @zwan0202 's changes won't appear in your PR diff

File: src/renderer/components/Editor/SquareBracket.tsx
Comment (ghost) [inline src/renderer/components/Editor/SquareBracket.tsx]: Please define this colour with other colours if possible. For reuse

File: src/renderer/components/Editor/SquareBracketHover.tsx
Comment (ghost) [inline src/renderer/components/Editor/SquareBracketHover.tsx]: Is this CSS logic needed as well as the JS logic used above?

File: src/renderer/components/Editor/WordComponent.tsx
Comment (ghost) [inline src/renderer/components/Editor/WordComponent.tsx:34]: is logic enabled all the time (for words that aren't in takes at all) or just for words in takes? (sorry I'm too blind to tell in the video)

File: src/renderer/components/Editor/TakeGroupComponent.tsx
Comment (ghost) [inline src/renderer/components/Editor/TakeGroupComponent.tsx]: Just checking that this CustomRowStack is necessary? It only seems to have 1 child component

File: src/renderer/components/Editor/TakeComponent.tsx
Comment (ghost) [inline src/renderer/components/Editor/TakeComponent.tsx]: Please extract these colours to variables for reusability

**Comment by ghost** [comment]
Love you work. Few tiny comments to peruse. 
.
As a token of my appreciation, here is your name with all letters replaced with sheep:
🐑🐑🐑🐑🐑        🐑🐑-🐑🐑🐑🐑🐑🐑

File: src/renderer/components/Editor/WordComponent.tsx
Comment (chloebrett) [reply src/renderer/components/Editor/WordComponent.tsx:34]: `isInInactiveTake` is false if a word is not in a take at all - for `isInInactiveTake` to be true, the word has to be (a) in a take and (b) that take has to be inactive

File: src/renderer/components/Editor/SquareBracket.tsx
Comment (chloebrett) [inline src/renderer/components/Editor/SquareBracket.tsx:14]: `isHovering` seems like it's only being used to adjust the opacity of the children, and it's a fair bit of JS logic to accomplish something that CSS is better suited to. Instead, you could define this in CSS. Something along the lines of (syntax might not be correct but the idea is there)

```
&:hover {
  & .innerBracket {
    opacity: 0.5;  
  }
}
```

Which would work if you used `styled()` to define the "squareBracket" box as we have for other components, and give the children a classname

**Review by chloebrett — COMMENTED** [review]
Agree with Hugh's comments, overall nice work

**Comment by chloebrett** [comment]
I changed your branch base to point to the multi sentence take detection one - makes the diff only show your changes, and you don't want to be merging this to develop anyway until the multi sentence take detection has landed. It doesn't mean that you should actually merge this into the MSTD branch, but instead that you should wait until that one is merged before merging this

File: src/renderer/components/Editor/SquareBracket.tsx
Comment (MutazAshhab) [reply src/renderer/components/Editor/SquareBracket.tsx:14]: I really liked your solution and I gave this a shot. The class names have to be unique so that it could only turn on the hover of SquareBracketHover component. If all the class names are the same then all the components will have the hover effect when you hover over 1.

I tried using a template literal so that I can have unique class names per SquareBracket component I got a syntax error.

Here is a screenshot: 
image


Im going to stash these changes and i'll show you them tomorrow

File: src/renderer/components/Editor/TakeGroupComponent.tsx
Comment (MutazAshhab) [reply src/renderer/components/Editor/TakeGroupComponent.tsx]: No, i just checked

With CustomRowStack:

image


Without CustomRowStack:

image

File: src/renderer/components/Editor/SquareBracketHover.tsx
Comment (MutazAshhab) [reply src/renderer/components/Editor/SquareBracketHover.tsx]: Nope

File: src/renderer/components/Editor/SquareBracket.tsx
Comment (chloebrett) [reply src/renderer/components/Editor/SquareBracket.tsx:14]: Very cool, you're so close 😅 

Because it's an object, if the key is a variable and not a string literal then you need to put it in square brackets. I'll give you a more generic example and you can apply to your specific situation

Example:

```javascript
const keyName = 'myKey';
const object = {
    [keyName]: 'someValue'
}
```

The above will produce this object: `{ "myKey": "someValue" }`
Whereas without the square brackets you get: `{ "keyName": "someValue" }`

This is because it's ambiguous otherwise whether you mean "I want my key to be the literal string 'keyName'" versus "I want my key to be whatever the value of the keyName variable is, which is 'myKey'"

Comment (MutazAshhab) [reply src/renderer/components/Editor/SquareBracket.tsx:14]: The javascript portion of that worked, 

image


But it is not hovering, when i use the dev tools it shows nothing up. 

image


I pushed those changes on another branch, I made a PR so you can see the diff: https://github.com/chloebrett/mlvet/pull/279. Let me know if you can spot whats wrong.

Code looks like this:

```javascript
  const hoverCss = { '&:hover': { [componentSelector]: { opacity: 0.5 } } };

  console.log(hoverCss);

  const SquareBracketBox = styled(Box)({
    height: '60px',
    width: '15px',
    borderStyle: 'solid',
    borderColor: isTakeGroupOpened ? colors.yellow[500] : colors.grey[500],
    borderWidth: '0px',
    borderLeftWidth: '2px',
    borderTopWidth: '2px',
    borderBottomWidth: bottomWidth,

    ...hoverCss,
  });
```

**Pull request closed without merging**
